### ISSUE TITLE
Support for different species

### ISSUE THREAD

**Issue by lardenoije** [MarWoes/wg-blimp#5]
Hi,

currently only the hg19 and hg38 human genome builds appear to be supported by this pipeline.
It would be great if support for other species could be added!
I'm particularly interested in using Mmul_10.

**Comment by MarWoes** [comment]
Definitely a good idea, I'll have a look at it!

**Comment by CraigMichell** [comment]
Hi, I would love to see this extended to non-model organisms.
For example being able to over-ride the genome reference and annotation with our own by including a fasta file and a GFF file.
Thanks for a great tool though!

**Comment by MarWoes** [comment]
Thanks for your comment! I agree that wg-blimp is too limited as of yet.

Firstly, I will likely simply add an option so omit annotation altogether. This way the whole workflow can at least be executed resulting in DMRs and segmentation. Currently, segmentation is not working without annotation, but unannotated DMRs can still be called by specifying in the configuration file:
```
target_files:
- dmr/combined-dmrs.csv
- qc/multiqc_report.html
- qc/methylation_metrics.csv
```

I'll have a look how to easiest integrate multiple species, but I think the best way would be to allow for arbitrary GFF/BED files, and provide some annotation tracks pre-built (for example for human data).

**Comment by MarWoes** [comment]
Implemented as of 7812ec0dbbe1c3f82ad557795104af6a8af0c0ab . mmul10 is added to the list of supported references. Gene annotations are now based on arbitrary GTF files, so in principle arbitrary organisms may now be run through the analysis pipeline.

GUI still needs some tweaks to not break if any annotation is missing, but I'll try to do some more in-depth testing and more bug fixing and include the changes in the next release, hopefully by the end of next week everything is updated.

For the time being, if there are any urgent analyses, `wg-blimp` may be installed through Bioconda, and then again installed from source using `pip install .` to include all recent changes

**Comment by lardenoije** [comment]
Thanks for adding this!

**Comment by MarWoes** [comment]
Included in release v0.9.6. If any issues remain, feel free to reopen!

**Comment by harish0201** [comment]
Is it possible to use Ensembl GTFs or GFFs? 

A step wherein conversion through the wonderful [gffread](https://github.com/gpertea/gffread) could be really helpful to have a consistent format!

**Comment by MarWoes** [comment]
Yes, Ensembl GTFs may be used for analysis. In fact, the included mmul10 setting will download the GTF files from Ensembl whereas human data uses Gencode GTF files. Thanks for the pointer, I'll have look if it's possible to support more formats!

**Comment by HeQSun** [comment]
Hi @MarWoes,

how it's going with this? 

Would it be possible for the pipeline to work with any other user-interested species now?

thanks,
Hequan

**Comment by MarWoes** [comment]
Hi @HeQSun ,

Thanks for reaching out! It is in theory possible to analyse arbitrary species with wg-blimp.

The only information you need is a reference fasta and a GTF file for gene annotation (Ensembl files should work). You can use the command `wg-blimp create-config --genome_build 'None'` to create a configuration file for your purposes. In the created YAML file you may then set the GTF parameter accordingly.

Please note that for methylome segmentation through MethylSeekR it is necessary to provide CpG island information.  These annotations were downloaded and put into this repository manually from UCSC, for example from [here](http://genome-euro.ucsc.edu/cgi-bin/hgTables?hgsid=271401961_3phaDsp9AbdEwNpYJqBqIHWOQb7f&clade=mammal&org=Human&db=0&hgta_group=allTracks&hgta_track=cpgIslandExt&hgta_table=cpgIslandExt&hgta_regionType=genome&position=&hgta_outputType=primaryTable&hgta_outFileName=). If you want to utilize segmentation, download the corresponding table from that resource, gzip it, and set the YAML config parameter accordingly.

There is currently only limited documentation for arbitrary species support, so I'll re-open this issue to look how to further document this use case.

I hope this helps a little bit, don't hesitate to reach out if you have any further questions or anything is breaking!

Best,
Marius

**Comment by HeQSun** [comment]
> Hi @HeQSun ,
> 
> Thanks for reaching out! It is in theory possible to analyse arbitrary species with wg-blimp.
> 
> The only information you need is a reference fasta and a GTF file for gene annotation (Ensembl files should work). You can use the command `wg-blimp create-config --genome_build 'None'` to create a configuration file for your purposes. In the created YAML file you may then set the GTF parameter accordingly.
> 
> Please note that for methylome segmentation through MethylSeekR it is necessary to provide CpG island information. These annotations were downloaded and put into this repository manually from UCSC, for example from [here](http://genome-euro.ucsc.edu/cgi-bin/hgTables?hgsid=271401961_3phaDsp9AbdEwNpYJqBqIHWOQb7f&clade=mammal&org=Human&db=0&hgta_group=allTracks&hgta_track=cpgIslandExt&hgta_table=cpgIslandExt&hgta_regionType=genome&position=&hgta_outputType=primaryTable&hgta_outFileName=). If you want to utilize segmentation, download the corresponding table from that resource, gzip it, and set the YAML config parameter accordingly.
> 
> There is currently only limited documentation for arbitrary species support, so I'll re-open this issue to look how to further document this use case.
> 
> I hope this helps a little bit, don't hesitate to reach out if you have any further questions or anything is breaking!
> 
> Best,
> Marius

Hi @MarWoes ,

thanks for your reply. I will try what you suggested, and let you know if it works..

Best,
Hequan
